### PR TITLE
Simple code refractor

### PR DESCRIPTION
**Pull request by aquapi**
Merge this if you like the changes

### CODE REVIEW

File: src/loader.ts [hunk 1]
```diff
@@ -31,9 +31,16 @@ export default class {
      * By default asks for the build command and output directory from the user on importing the source file for the first time.
      */
     async initConfigPre() {
-        console.log(`\x1b[33m[HYPERIMPORT]\x1b[39m: ${this.name}\nNo configuration was found for "${this.config.importPath}"\nEnter the build command and output directory to configure it.\nPress enter to use the default values.\n`);
+        console.log(
```
Comment (tr1ckydev) [inline src/loader.ts:34]: Thanks, that makes it easy to read!

File: src/loader.ts [hunk 2]
```diff
@@ -42,29 +49,43 @@ export default class {
      */
     async initConfigTypes() {
         const filename = basename(this.config.importPath);
+        
         mkdirSync(`${this.cwd}/@types/${filename}`, { recursive: true });
         Bun.write(`${this.cwd}/@types/${filename}/lastModified`, lastModified(this.config.importPath));
+
```
Comment (tr1ckydev) [inline src/loader.ts:57]: I am opposed to having a blank line here which essentially means the above two statements are grouped but they shouldn't be.

File: src/loader.ts [hunk 3]
```diff
@@ -42,29 +49,43 @@ export default class {
...
         const configWriter = Bun.file(`${this.cwd}/@types/${filename}/config.ts`).writer();
-        configWriter.write(`import { LoaderConfig, T } from "hyperimport";\nexport default {\n\tbuildCommand: ${JSON.stringify(this.config.buildCommand)},\n\toutDir: "${this.config.outDir}",\n\tsymbols: {`);
-        for (const symbol of nm(this.config.libPath)) {
+        
+        configWriter.write(
+            `import { LoaderConfig, T } from "hyperimport";\n` 
+            + `export default {\n\tbuildCommand: ${JSON.stringify(this.config.buildCommand)},\n\toutDir: "${this.config.outDir}",\n\tsymbols: {`
+        );
+        
+        for (const symbol of nm(this.config.libPath))
```
Comment (tr1ckydev) [inline src/loader.ts:65]: lgtm!

File: src/loader.ts [hunk 4]
```diff
@@ -42,29 +49,43 @@ export default class {
...
 
     /**
      * When the source file isn't configured yet, this executes to configure it.
      */
     async initConfig() {
         await this.initConfigPre();
+        
         console.log("\nBuilding the source file...");
         await this.build();
+
```
Comment (tr1ckydev) [inline src/loader.ts:89]: I am opposed for this new line here too, the source file build success message should be together with them.

File: src/loader.ts [hunk 5]
```diff
@@ -73,8 +94,9 @@ export default class {
      * Checks if the source file was modified, if it is, then `build()` is executed to rebuild the changed source file.
      */
     async ifSourceModify() {
-        const lm = lastModified(this.config.importPath);
-        const lmfile = `${this.cwd}/@types/${basename(this.config.importPath)}/lastModified`;
+        const lm = lastModified(this.config.importPath),
+            lmfile = `${this.cwd}/@types/${basename(this.config.importPath)}/lastModified`;
```
Comment (tr1ckydev) [inline src/loader.ts]: This is a chaotic change and takes away readability. I would suggest reverting them.

File: src/loader.ts [hunk 6]
```diff
@@ -110,20 +132,20 @@ export default class {
...
     async toPlugin(): Promise<BunPlugin> {
-        const parentThis = this;
         return {
-            name: parentThis.name,
-            setup(build) {
-                build.onLoad({ filter: new RegExp(`\.(${parentThis._config.extension})$`) }, async args => {
-                    parentThis.config.importPath = args.path;
-                    await parentThis.preload();
+            name: this.name,
+            // Arrow function does not have a scope so it can access parent this
```
Comment (tr1ckydev) [inline src/loader.ts]: The only reason I had to use `parentThis` because typescript auto completions weren't working with `this` inside function scope.

File: preload.ts [hunk 7]
```diff
@@ -1,23 +1,30 @@
 import { HyperImportConfig } from "./src/types";
 import { debugLog } from "./src/utils";
 
-const cwd = process.cwd();
-const config: HyperImportConfig = (await import(`${cwd}/bunfig.toml`)).default.hyperimport;
+const cwd = process.cwd(),
+    config: HyperImportConfig = (await import(`${cwd}/bunfig.toml`)).default.hyperimport;
```
Comment (tr1ckydev) [inline preload.ts]: Same with here too.

File: preload.ts [hunk 8]
```diff
@@ -1,23 +1,30 @@
...
 
 debugLog(config.debug, 3, "registering loaders...");
 
-for (const loader of config.loaders ?? []) {
-    await import(`./src/loaders/${loader}.ts`).then(async l => {
-        const plugin = await new l.default(cwd).toPlugin();
+for (const loader of config.loaders ?? []) 
+    try {
```
Comment (tr1ckydev) [inline preload.ts]: Adding a catch function directly when awaiting the import makes it easier to read and is concise too. I would recommend to revert this change.

File: preload.ts [hunk 9]
```diff
@@ -1,23 +1,30 @@
...
+    } catch (e) {
+        debugLog(config.debug, 1, "loader not found:", loader);
+    }
 
-for (const loader of config.custom ?? []) {
-    await import(`${cwd}/${loader}`).then(async l => {
-        const plugin = await new l.default(cwd).toPlugin();
+
+for (const loader of config.custom ?? []) 
+    try {
```
Comment (tr1ckydev) [inline preload.ts]: Same with here too.

**Review by tr1ckydev — CHANGES_REQUESTED** [review]

File: src/loader.ts (hunk 6 above)
Comment (ImLunaHey) [reply src/loader.ts]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions

**Comment by aquapi** [comment]
@tr1ckydev I made some changes
